### PR TITLE
Refactor DateTime into Date and Time

### PR DESCRIPTION
**Pull request by SimWPEric**
We need `Appointment` to hold 2 `DateTime` to indicate start and end of  `Appointment`.
But since the start and end will be on the same day, we do not need to store date twice. 
However, the `DateTime` class currently includes both `Date` and `Time`.
`DateTime` is refactored into `Date` and `Time` for `Appointment` to have:
1. `Date date;` 
2. `Time startTime;`
3. `Time endTime;`

This change also makes it easier to implement features like filter by date and check Appointment overlap.

### CODE REVIEW

File: src/main/java/seedu/address/model/appointment/Date.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        }
+
+        // April, June, September, November have 30 days
+        if (month == 4 || month == 6 || month == 9 || month == 11) {
+            if (day > 30) {
+                return false;
+            }
+        }
+
+        if (month == 2) {
```
Comment (quzhetao01) [inline src/main/java/seedu/address/model/appointment/Date.java:63]: Good comprehensive checking here

File: src/main/java/seedu/address/model/appointment/Appointment.java [hunk 2]
```diff
@@ -43,7 +61,20 @@ public Description getDescription() {
...
 
     /**
-     * Returns true if both appointments have the same date and time.
+     * Checks the validity of the start and end times of this appointment.
+     *
+     * @param startTime The start time of the appointment.
+     * @param endTime   The end time of the appointment.
+     * @throws InvalidStartEndTimeException If the start time is not before the end time, an exception is thrown.
+     */
+    public void checkStartEndTime(Time startTime, Time endTime) throws InvalidStartEndTimeException {
```
Comment (Sheeepen) [inline src/main/java/seedu/address/model/appointment/Appointment.java:70]: good use of Exception!

**Review by Sheeepen — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/model/appointment/Date.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    public final LocalDate localDate;
+
+    /**
+     * Constructs a {@code Date}.a
+     *
+     * @param date A valid date.
+     */
+    public Date(String date) {
+        requireNonNull(date);
+        checkArgument(isValidDate(date), MESSAGE_CONSTRAINTS);
```
Comment (quzhetao01) [inline src/main/java/seedu/address/model/appointment/Date.java:30]: I think MESSAGE_CONSTRAINTS should be tweaked slightly based on isValidDate() implementation

File: src/main/java/seedu/address/model/util/DateTimeParser.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        return localTime.format(OUTPUT_TIME_FORMATTER);
+    }
+
+    /**
+     * Formats the given date string to a more readable format.
+     *
+     * @param date The date string in "yyyy-MM-dd" format to be formatted.
+     * @return A formatted date string in "dd MMMM yyyy" format (e.g., 31 December 2023).
+     */
+    public static String formatDate(String date) {
```
Comment (quzhetao01) [inline src/main/java/seedu/address/model/util/DateTimeParser.java:48]: useful method

**Review by quzhetao01 — COMMENTED** [review]
LGTM! Some minor tweaks is needed bah

**Pull request merged**